keras/layers/pooling.py

```diff
@@ -424,7 +444,21 @@ class GlobalAveragePooling1D(_GlobalPooling1D):
     '''
 
     def call(self, x, mask=None):
-        return K.mean(x, axis=1)
+        if mask is None:
+            return K.mean(x, axis=1)
+        else:
+            mask = K.cast(mask, K.floatx())
+            # Expand count so we can do a broadcast division
+            count = K.sum(mask, axis=1)
+            count = K.expand_dims(count, dim=1)
+            if K.backend() == 'tensorflow':  # tensorflow shape command dosn't work
```

Comment:
`int_shape` only exists for TF because only TF implements it. Future backends will hopefully implement it, and maybe Theano will implement it too in the future.

Another reason why this implementation won't work: these layers are meant to work for variable-size inputs (that's basically the motivation for having them vs. using regular pooling layers). So you cannot access `int_shape`, it won't work with None dimensions.

You will have to reimplement this to make it work with symbolic shapes.